### PR TITLE
Remove join between Artifact and ArtifactCollection table

### PR DESCRIPTION
**Pull request by serinamarie**
OSS implementation of: https://github.com/PrefectHQ/nebula/pull/4303

Adds a route `/artifacts/collections/filter` for ability to filter the ArtifactCollection table the same way as the artifact table. This allows the initial page, which displays the latest of each artifact, to be filtered by flow run, etc. This means we won't need to join the tables.
- We needed a new method called `read_latest_artifacts` that is used by the new route.
- Added and updated tests accordingly

The model methods needed to be updated to upsert all data (not just id and key) to the `artifact_collection` table.
- Updated tests accordingly

Migrates `artifact` table data to `artifact_collection` if it is the latest version of the artifact.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: src/prefect/server/schemas/filters.py [hunk 1]
```diff
@@ -1557,51 +1557,167 @@ def _get_filter_list(self, db: "PrefectDBInterface") -> List:
...
+            "If `true`, only include artifacts with a non-null key. If `false`, "
+            "only include artifacts with a null key. Should return all rows in "
+            "the ArtifactCollection table if specified."
         ),
     )
 
     def _get_filter_list(self, db: "PrefectDBInterface") -> List:
         filters = []
-        if self.is_latest:
-            filters.append(db.ArtifactCollection.latest_id == db.Artifact.id)
```
Comment (serinamarie) [inline src/prefect/server/schemas/filters.py:1574]: 🥾

**Review by jakekaplan — APPROVED** [review]
🙌

File: src/prefect/server/schemas/core.py [hunk 2]
```diff
@@ -1203,3 +1203,42 @@ def validate_metadata_length(cls, v):
             if len(str(v[key])) > max_metadata_length:
                 v[key] = str(v[key])[:max_metadata_length] + "..."
         return v
+
+
+class ArtifactCollection(ORMBaseModel):
+    key: str = Field(
+        default=None, description="An optional unique reference key for this artifact."
```
Comment (serinamarie) [inline src/prefect/server/schemas/core.py:1210]:
```suggestion
        description="An optional unique reference key for this artifact."
```

File: src/prefect/server/schemas/core.py [hunk 3]
```diff
@@ -1203,3 +1203,42 @@ def validate_metadata_length(cls, v):
             if len(str(v[key])) > max_metadata_length:
                 v[key] = str(v[key])[:max_metadata_length] + "..."
         return v
+
+
+class ArtifactCollection(ORMBaseModel):
+    key: str = Field(
+        description="An optional unique reference key for this artifact."
+    )
```
Comment (serinamarie) [inline src/prefect/server/schemas/core.py:1211]:
```suggestion
    key: str = Field(description="An optional unique reference key for this artifact.")
```

**Pull request merged**